### PR TITLE
add helper function for v20160930

### PR DESCRIPTION
**Pull request by rjtsdl**
I just mimic the helpers for v20160930. 
I am still looking how to test the correctness of those changes. I don't see tests for versioned apis.

### CODE REVIEW

**Comment by rjtsdl** [comment]
@anhowe 

i have added testData for v20160930. And resolved the conflicts as well

File: pkg/api/converterfromapi.go [hunk 1]
```diff
@@ -111,6 +154,16 @@ func convertPropertiesToVLabs(api *Properties, vlabsProps *vlabs.Properties) {
...
 }
 
+func convertLinuxProfileToV20160930(api *LinuxProfile, v20160930 *v20160930.LinuxProfile) {
+	v20160930.AdminUsername = api.AdminUsername
+	v20160930.SSH.PublicKeys = []struct {
+		KeyData string `json:"keyData"`
+	}{}
+	for _, d := range api.SSH.PublicKeys {
+		v20160930.SSH.PublicKeys = append(v20160930.SSH.PublicKeys, d)
+	}
```
Comment (amanohar) [inline pkg/api/converterfromapi.go]: This is different from what's there in: convertLinuxProfileToV20160330.
Do these need to be different and why?

File: pkg/api/convertertoapi.go [hunk 2]
```diff
@@ -226,17 +321,43 @@ func convertV20160330VMDiagnostics(v20160330 *v20160330.VMDiagnostics, api *VMDi
...
 	api.ClientID = vlabs.ClientID
 	api.Secret = vlabs.Secret
 }
 
+func convertV20160930CustomProfile(v20160930 *v20160930.CustomProfile, api *CustomProfile) {
+	api.Orchestrator = v20160930.Orchestrator
+}
+
+func convertV20160930CertificateProfile(api *CertificateProfile) {
+	api.CaCertificate = "caCertificate"
```
Comment (anhowe) [inline pkg/api/convertertoapi.go]: leave these empty string "".

File: pkg/api/converterfromapi.go [hunk 3]
```diff
@@ -153,6 +211,10 @@ func convertWindowsProfileToVLabs(api *WindowsProfile, vlabsProfile *vlabs.Windo
 	}
 }
 
+func convertOrchestratorProfileToV20160930(api *OrchestratorProfile, o *v20160930.OrchestratorProfile) {
+	o.OrchestratorType = v20160930.OrchestratorType(api.OrchestratorType)
```
Comment (anhowe) [inline pkg/api/converterfromapi.go]: convert DCOS187 to DCOS here

File: pkg/api/v20160930/validate.go [hunk 4]
```diff
@@ -10,6 +10,7 @@ import (
 func (o *OrchestratorProfile) Validate() error {
 	switch o.OrchestratorType {
 	case DCOS:
+	case DCOS187:
```
Comment (anhowe) [inline pkg/api/v20160930/validate.go]: remove this

File: pkg/api/v20160930/const.go [hunk 5]
```diff
@@ -10,7 +10,8 @@ const (
 	// Mesos is the string constant for the Mesos orchestrator type
 	Mesos OrchestratorType = "Mesos"
 	// DCOS is the string constant for DCOS orchestrator type and defaults to DCOS187
-	DCOS = "DCOS"
+	DCOS    = "DCOS"
+	DCOS187 = "DCOS187"
```
Comment (anhowe) [inline pkg/api/v20160930/const.go]: remove this

File: pkg/api/converterfromapi.go (hunk 1 above)
Comment (rjtsdl) [inline pkg/api/converterfromapi.go]: They should be the same. Btw i didn't see any difference

**Pull request merged**